msteams-platform/toolkit/add-API-connection.md

```diff
@@ -0,0 +1,202 @@
+   |Azure AD|teamsfx add api-connection aad --endpoint <https://example.com> --component bot --alias example --app-type custom --tenant-id your_tenant_id --app-id your_app_id --interactive false|
+   |Certificate|teamsfx add api-connection cert --endpoint <https://example.com> --component bot --alias example --interactive false|
+   |Custom|teamsfx add api-connection custom --endpoint <https://example.com> --component bot --alias example --interactive false|
+
+## Understand Toolkit updates to your project
+
+ Teams Toolkit modifies `bot` or `api` folder based on your selections:
+
+1. Generate `{your_api_alias}.js/ts` file. The file initializes an API client for your API and exports the API client.
+
+2. Add `@microsoft/teamsfx` package to `package.json`. The package provides support for the common API authentication methods.
+
+3. Add environment variables to `.env.teamsfx.local`. They are the configurations for the selected authentication type. The generated code reads values from the environment variables.
+
+## Test API connection in local environment
+
+The following steps help to test the API connection in the Teams Toolkit local environment:
+
+ 1. **Run npm install**
+
+    Run `npm install` under `bot` or `api` folder to install added packages.
+
+ 2. **Add API credentials to the local application settings**
+
+    Teams Toolkit does't ask for credentials but it leaves placeholders in the local application settings file. Replace the placeholders with the appropriate credentials to access the API. The local application settings file is the `.env.teamsfx.local` file in the `bot` or `api` folder.
+
+ 3. **Use the API client to make API requests**
+
+    Import the API client from the source code that needs access to the API:
+
+    ```BASH
+    import { yourApiClient } from '{relative path to the generated file}'
+    ```
+
+ 4. **Generate http(s) requests to target API (with Axios)**
+
+    The generated API client is an Axios API client. Use the Axios client to make requests to the API.
+
+     > [!Note]
+     >[Axios](https://www.npmjs.com/package/axios) is a popular nodejs package that helps you with http(s) requests. For more information on how to make http(s) requests, see [Axios example documentation](https://axios-http.com/docs/example) to learn how to make http(s).
+
+## Deploy your application to Azure
+
+To deploy your application to Azure, you need to add the authentication to the application settings for the appropriate environment. For example, your API might have different credentials for `dev` and `prod`. Based on environment needs, configure Teams Toolkit.
+
+Teams Toolkit configures your local environment. The bootstrapped sample code contains comments that tell you what app settings you need to configure. For more information on application settings, see [Add app settings](https://github.com/OfficeDev/TeamsFx/wiki/%5BDocument%5D-Add-app-settings).
+
+## Advanced scenarios
+
```

Comment:
Can I suggest we use below style for advanced scenarios section? 
![image](https://user-images.githubusercontent.com/11220663/169492202-94abc4ba-ee9e-46c8-ba8e-62fa5bbccfda.png)

Reply:
Updated as commented